pkg/controller/cloud/nodecontroller.go

```diff
@@ -55,6 +65,13 @@ const (
 
 	// The amount of time the nodecontroller should sleep between retrying NodeStatus updates
 	retrySleepTime = 20 * time.Millisecond
+
+	//Taint denoting that a node needs to be processed by external cloudprovider
+	CloudTaintKey = "ExternalCloudProvider"
+
+	nodeStatusUpdateFrequency = 10 * time.Second
+
+	LabelProvidedIPAddr = "beta.kubernetes.io/provided-node-ip"
```

Comment:
nodeIP is an option to specify the IP address of a node when starting Kubelet. The kubelet verifies that this nodeIP is  valid in the cloudprovider.